cupyx/jit/_builtin_funcs.py

```diff
@@ -50,6 +51,50 @@ def call(self, env, *args, **kwargs):
         return Range(start, stop, step, ctype, step_is_positive)
 
 
+class Len(BuiltinFunc):
+    def call(self, env, *args, **kwds):
+        if len(args) > 1:
+            raise TypeError(f'len expects only 1 argument, got {len(args)}')
+        arg = args[0]
+        if isinstance(arg.ctype, _cuda_types.CArray):
+            return Data(f'{arg.code}.size()', _cuda_types.PtrDiff())
+        if isinstance(arg.ctype, _cuda_types.SharedMem):
+            if arg.ctype._size is None:
+                raise TypeError('unsized shared memory is not supported')
+            return Constant(arg.ctype._size)
```

Comment:
The variable of SharedMem type is not exposed to the users, because the return value of `shared_memory()` is converted to Ptr type.
https://github.com/cupy/cupy/blob/1af3300bfaa224d5f0da8f8a2a0daf91f9c406f8/cupyx/jit/_builtin_funcs.py#L123-L129

Reply:
You are right. I overlooked that part...